src/main/scala/pureconfig/ConfigFieldMapping.scala

```diff
@@ -0,0 +1,24 @@
+package pureconfig
+
+trait ConfigFieldMapping[T] {
```

Comment:
Well, easier said than done. I think I won't be able to provide an implementation for my proposal until tomorrow, sorry.